BluePic-Server/Sources/Server/RouteUtils.swift

```diff
@@ -53,42 +53,37 @@ func processImage(withId imageId: String) {
 
   // Make REST call
   let req = HTTP.request(requestOptions) { resp in
-    if let resp = resp where resp.statusCode == HTTPStatusCode.OK || resp.statusCode == HTTPStatusCode.accepted {
+    if let resp = resp , resp.statusCode == HTTPStatusCode.OK || resp.statusCode == HTTPStatusCode.accepted {
       do {
-        let body = NSMutableData()
-        try resp.readAllData(into: body)
+        var body = Data() //NSMutableData()
+        try resp.readAllData(into: &body)
         let jsonResponse = JSON(data: body)
         print("OpenWhisk response: \(jsonResponse)")
       } catch {
         Log.error("Bad JSON document received from OpenWhisk.")
       }
     } else {
-      Log.error("Status error code or nil reponse received from OpenWhisk.")
-      if let resp = resp {
-        Log.error("Status code: \(resp.statusCode)")
-        if let rawUserData = try? BodyParser.readBodyData(with: resp) {
-          #if os(OSX)
-            let str = NSString(data: rawUserData as Data, encoding: String.Encoding.utf8.rawValue)
-          #else
-            let str = NSString(data: rawUserData, encoding: NSUTF8StringEncoding)
-          #endif
-          print("Error response from OpenWhisk: \(str)")
+        Log.error("Status error code or nil reponse received from OpenWhisk.")
+        if let resp = resp {
+            Log.error("Status code: \(resp.statusCode)")
+            var rawUserData = Data()
+            do {
+                try resp.read(into: &rawUserData)  //try? BodyParser.readBodyData(with: resp) {
```

Comment:
We will test it and see if it works as we are expecting. Looking at the Kitura code, it seems that the code above should work. Though it is odd that the signature for BodyParser.readBodyData() has changed.